src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -504,6 +531,12 @@ def getBlock(self, newFile, location, skipOpenCheck=False):
         newBlock = DBSBufferBlock(name=blockname,
                                   location=location,
                                   datasetpath=datasetpath)
+
+        parent = self.datasetParentageCache.get(datasetpath)
+        if parent:
+            newBlock.addDatasetParent(parent)
+            logging.info("Get block: Child dataset %s, Parent dataset %s", blockInfo['datasetpath'], parent)
```

Comment:
Just to make it consistent, how about using `datasetpath` instead of `blockInfo['datasetpath']`?